scripts/ci/check_compliance.py

```diff
@@ -658,6 +659,32 @@ def check_no_undef_within_kconfig(self, kconf):
         if undef_ref_warnings:
             self.failure(f"Undefined Kconfig symbols:\n\n {undef_ref_warnings}")
 
+    def check_soc_name_sync(self, kconf):
+        root_args = argparse.Namespace(**{'soc_roots': [Path(ZEPHYR_BASE)]})
+        v2_systems = list_hardware.find_v2_systems(root_args)
+
+        soc_names = {soc.name for soc in v2_systems.get_socs()}
+
+        soc_kconfig_names = set()
+        for node in kconf.node_iter():
+            if isinstance(node.item, kconfiglib.Symbol) and node.item.name == "SOC":
```

Comment:
might want to add `# pylint: disable=undefined-variable` to suppress the complaint by pylint.

Reply:
Done.